.changeset/light-sheep-give.md

```diff
@@ -0,0 +1,26 @@
+---
+"remix": patch
+"@remix-run/serve": patch
+"@remix-run/server-runtime": patch
+---
+
+Fix `TypedResponse` so that Typescript correctly shows errors for incompatible types in loaders and actions.
+
+Previously, when the return type of a loader or action was explicitly set to `TypedResponse<SomeType>`,
+Typescript would not show errors when the loader or action returned an incompatible type.
+
+For example:
+
+```ts
+export const action = async (args: ActionArgs): Promise<TypedResponse<string>> => {
+    return json(42);
+};
+```
+
+In this case, Typescript would now show an error even though `42` is clearly not a `string`.
```

Comment:
I think we want "In this case, Typescript would *not* show an error..." (now -> not)

Reply:
I can make the change real quick